ui/src/app/common/services/orb.service.ts

```diff
@@ -103,45 +101,27 @@ export class OrbService implements OnDestroy {
     this.killPolling = new Subject<void>();
 
     this.pollController$ = new BehaviorSubject<boolean>(PollControls.PAUSE);
+  }
 
-    const controller = merge(
-      this.pollController$.pipe(
-        switchMap((control) => {
-          if (control === PollControls.RESUME)
-            return defer(() => timer(1, this.pollInterval));
-          return EMPTY;
-        }),
-      ),
-      this.forceRefresh.pipe(debounceTime(1000)),
-    );
-
-    this.poller$ = controller.pipe(takeUntil(this.killPolling));
-
-    /**
-     * TODO turn orb service into a poller service
-     * available in root, inject it into a @Observe decorator
-     * instead and wrap desired observables in it
-     */
-
-    this.agents$ = this.observe(this.agent.getAllAgents());
-
-    this.groups$ = this.observe(
-      this.group.getAllAgentGroups(),
-    );
-
-    this.policies$ = this.observe(
-      this.policy.getAllAgentPolicies(),
-    );
+  private mapTags = (list: AgentGroup[] & Sink[]) => {
+    return list
+      .map((item) =>
+        Object.entries(item.tags).map((entry) => `${entry[0]}: ${entry[1]}`),
+      )
+      .reduce((acc, val) => acc.concat(val), [])
+      .filter(this.onlyUnique);
+  }
 
-    this.datasets$ = this.observe(
-      this.dataset.getAllDatasets(),
-    );
+  ngOnDestroy() {
+    this.killPolling.next();
+  }
 
-    this.sinks$ = this.observe(
-      this.sink.getAllSinks(),
-    );
+  getAgentListView() {
+    return this.observe(this.agent.getAllAgents());
+  }
 
-    this.agentsTags$ = this.agents$.pipe(
+  getAgentsTags() {
+    return this.observe(this.agent.getAllAgents()).pipe(
```

Comment:
once observe() is made into a decorator, I can stop wrapping it around the observed thing and simply add @observe on top of each method.